g1ant-language/desktop-apps-automation2.md

```diff
@@ -0,0 +1,211 @@
+
+Take emails, for example. You don’t have to use any external tool such as email client or net browser to check your inbox: there is the `mail.imap` command for it. You simply provide your email account credentials as the arguments and the command gets messages for you, storing them in a variable. Then, you can search this variable for some text and even replace it with the `text.` family of commands.  And this is exactly what you are going to do in this exercise.
+
+### Reading Emails
+
+The whole process described in your plan of actions can be divided into three main tasks: retrieving emails, searching for specific messages and deleting email addresses from the list. Let’s make procedures out of these tasks and name them `checkmail`, `checksubject` and `deleteaddress`.
+
+```G1ANT
+procedure checkmail
+    
+end procedure
+
+procedure checksubject
+    
+end procedure
+
+procedure deleteaddress
+    
+end procedure
+```
+
+Start with the `checkmail` procedure. It should simply login to your email account and retrieve messages. The `mail.imap` command does it in one fell swoop. What it needs is the email server information, your login credentials and a name of the variable, which will store emails. You can also specify some filtering, which is helpful when processing huge mailboxes: time range (start and end dates of messages), an option to process only unread messages and whether to mark them as read afterwards. The full syntax for the command is as follows:
+
+```G1ANT
+mail.imap host ‴imap.server.name‴ port ‴993‴ login ‴your@email.address‴ password ‴p@$$w0rd‴ sincedate ♥startdate todate ♥enddate onlyunreadmessages true markallmessagesasread true result ♥result
+```
+
+Let’s assume your script should check only unread messages received since the day before the robot is run. These messages will be marked as read (it’s the default value, so you don’t have to specify it) and will be passed to the `♥emails` variable. So here is your `checkmail` procedure:
+
+```G1ANT
+procedure checkmail
+	mail.imap host ‴imap.server.name‴ port ‴993‴ login ‴your@email.address‴ password ‴p@$$w0rd‴ sincedate ♥yesterday onlyunreadmessages true result ♥emails
+end procedure
+```
+
+Of course, you should provide your true server name, login and password in the code above.
+
+There are two variables used in this procedure: `♥yesterday` and `♥emails`. The latter is created automatically by the command (the default name is `♥result`), but the first needs to be declared beforehand. Let’s do it.
```

Comment:
the first one has to be a variable, since we create it with the C# snippet, don't we?
The latter IS created automatically, because you don't have to declare it beforehand. You only choose not to use its default name.

Reply:
okay I agree :D